lib/imagemounter/index.js

```diff
@@ -0,0 +1,119 @@
+        }
+        if (!await exists(imageSignatureFilePath)) {
+            throw new Error(`The signature file provided not exists: ${imageFilePath}`);
+        }
+        if ((await lstat(imageSignatureFilePath)).isDirectory()) {
+            throw new Error(`The given signature file is expected to be a file, a directory was given: ${imageFilePath}`);
+        }
+        //read signature
+        const signature = await new B((resolve, reject) => {
+            readFile(imageSignatureFilePath, (err, data) => {
+                if (err != null) {
+                    reject(err);
+                } else {
+                    resolve(data);
+                }
+            });
+        });
+        const mountedImages = await this.lookup(imageType);
+        if (mountedImages.find((mountedSignature) => signature.equals(mountedSignature))) {
+            log.info(`An image with same signature of ${imageSignatureFilePath} is mounted. Doing nothing here`);
+            return;
+        }
+        //notify device
+        const imageSize = imageFileStat.size;
+        let ret = await this._plistService.sendPlistAndReceive({
+            Command: 'ReceiveBytes',
+            ImageSignature: signature,
+            ImageSize: imageSize,
+            ImageType: imageType
+        });
+        this._checkReturnError(ret);
+        if (ret.Status !== 'ReceiveBytesAck') {
+            throw new Error(`Unexpected return from ${MOBILE_IMAGE_MOUNTER_SERVICE_NAME} on sending ReceiveBytes: ${JSON.stringify(ret)}`);
+        }
+        //push image to device
+        const stream = createReadStream(imageFilePath);
+        try {
+            await new B((resolve, reject) => {
+                stream.on('data', async (data) => {
+                    await this._plistService._socketClient.write(data);
+                });
+                stream.on('end', resolve);
+                stream.on('error', reject);
+            });
+        } finally {
+            stream.close();
+        }
+        ret = await this._plistService.receivePlist();
+        this._checkReturnError(ret);
+        if (ret.Status !== 'Complete') {
+            throw new Error(`Unexpected return from ${MOBILE_IMAGE_MOUNTER_SERVICE_NAME} on pushing image file: ${JSON.stringify(ret)}`);
+        }
+        //mount image
+        ret = await this._plistService.sendPlistAndReceive({
+            Command: 'MountImage',
+            ImagePath: '/private/var/mobile/Media/PublicStaging/staging.dimag',
+            ImageSignature: signature,
+            ImageType: imageType
+        });
+        if (ret.DetailedError?.includes('is already mounted at /Developer')) {
+            log.warn('DeveloperImage was mounted');
+            return;
+        }
+        this._checkReturnError(ret);
+    }
+
+}
+export { ImageMounter, MOBILE_IMAGE_MOUNTER_SERVICE_NAME };
+export default ImageMounter;
```

Comment:
I know we `export default` stuff that's already exported everywhere, but we really don't need to.

```suggestion
```